### PR TITLE
Add a way to combine groups of variables

### PR DESCRIPTION
**Pull request by ebrelsford**
Add a function for combining common variables and an example of how this would be used.

The concept is that you would define files of variables that are JS modules with a default export of an object of variables. This change provides a function to combine these common variables with overrides specific to a particular style.

Nested variables are supported.

### CODE REVIEW

File: examples/shared-variables/templates/styles/style1.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const { combineVariables } = require('../../../../dist/main');
+
+module.exports.context = {
+  sources: {
+    root: "Example"
+  },
+  colors: combineVariables(sharedColors, {
+    road: 'gray'
+  })
+};
```
Comment (ebrelsford) [inline examples/shared-variables/templates/styles/style1.js:11]: This is an example of how one would re-use common colors--define them in a separate file, import them, then combine them using `combineVariables()`. In this case, the style uses all of the `sharedColors` but overrides `road`.

File: examples/shared-variables/templates/variables/colors.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  backgroundLow: 'red',
+  backgroundHigh: 'blue',
+  road: 'green'
+};
```
Comment (ebrelsford) [inline examples/shared-variables/templates/variables/colors.js:5]: Here is how we add common color variables.

File: src/lib/__test__/combine-variables.test.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        roads: {
+          major: 'yellow',
+          minor: 'white'
+        }
+      },
+      {
+        background: { high: 'green' } // Just tweak one nested property
+      }
+    );
```
Comment (ebrelsford) [inline src/lib/__test__/combine-variables.test.js:50]: Nested variables are supported and you can choose specific nested variables to override, in this case `background.high` is being overridden while the rest of the variables remain the same.

**Comment by ebrelsford** [comment]
@kelsey-taylor I'm curious to hear if you think this would fit our needs or if there are cases I'm missing!

**Review by kelsey-taylor — APPROVED** [review]
yep, this makes sense to me! i love how clean this is and how few changes it will take to implement it in https://github.com/stamen/amazon-MapStyleSandbox 🙏

**Pull request merged**